CodeU_Assignement_2/src/codeu_assignement_2/KemBinaryTree.java

```diff
@@ -0,0 +1,394 @@
+package codeu_assignement_2;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Stack;
+import java.util.Queue;
+import java.util.LinkedList;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Binary tree with a tree node
+ * Each node points to parent, and left/right child
+ * 
+ */
+public class KemBinaryTree <E> {
+    /* Data members of Tree */
+    protected KemTreeNode<E> m_root=null;
+    
+    /* */
+    public KemBinaryTree(){} 
+            
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_root == null); }
+    
+    /* Allow us to decide where to add items */
+    public KemTreeNode<E> getRoot() { return m_root;}
+    
+    /* Q1: Get all ancestors */
+    public void printAncestors(KemTreeNode<E> child)
+    {  
+        /* Check if child isn't null */
+        if (child == null)
+        {
+            System.out.println("Error: ** Null Pointer **");
+            return;
+        }
+        
+        /* Check if child isn't root */
+        KemTreeNode<E> curr = child.getParent();
+        if (curr == null)
+        {
+            System.out.println("** Root **: has no ancestors. Done.");
+            return;
+        }
+        
+        /* Print all ancestors as long as didn't get to root */
+        while (curr != null)

```

Comment:
I think this would be fine - it seems to be consistent with the assignment request.